File: Backend/src/main/java/com/bachld/project/backend/dto/response/detai/DeTaiResponse.java
package com.bachld.project.backend.dto.response.detai;

import com.bachld.project.backend.enums.DeTaiState;
import lombok.*;
import lombok.experimental.FieldDefaults;

@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = lombok.AccessLevel.PRIVATE)
public class DeTaiResponse {
    Long id;
    String tenDeTai;
    DeTaiState trangThai;
    String nhanXet;

    Long gvhdId;
    String gvhdTen;
    Long sinhVienId;

    String tongQuanDeTaiUrl;
    String tongQuanFilename;
}

File: Android/app/src/main/java/com/bachld/android/data/dto/response/DeTaiResponse.kt
package com.bachld.android.data.dto.response

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class DeTaiResponse(
    @Json(name = "id") val id: Long,
    @Json(name = "tenDeTai") val title: String,
    @Json(name = "trangThai") val status: String,
    @Json(name = "nhanXet") val review: String? = null,
    @Json(name = "gvhdId") val advisorId: Long,
    @Json(name = "gvhdTen") val advisorName: String?,
    @Json(name = "sinhVienId") val studentId: Long? = null,
    @Json(name = "tongQuanDeTaiUrl") val overviewUrl: String? = null
)